### PR TITLE
simplify arch for preload library

### PR DESCRIPTION
**Pull request by kasperk81**

### CODE REVIEW

**Comment by kasperk81** [comment]
@marcpopMSFT tiny change removing a hack

**Comment by marcpopMSFT** [comment]
Enum.GetName may not be the ideal solution so I'd prefer to avoid that (Daniel believes that'll use reflection). We don't want every value from RuntimeEnformation but is there a simpler way of mapping this onto the three values we do want?

File: src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs [hunk 1]
```diff
@@ -39,8 +39,7 @@ static Interop()
         private static void PreloadWindowsLibrary(string dllFileName)
         {
             string basePath = Path.GetDirectoryName(typeof(Interop).Assembly.Location);
-            string architecture = IntPtr.Size == 8 ? "x64" : "x86";
-            architecture = RuntimeInformation.ProcessArchitecture == Architecture.Arm64 ? "arm64" : architecture;
+            string architecture = Enum.GetName(typeof(Architecture), RuntimeInformation.ProcessArchitecture).ToLowerInvariant();
             string dllPath = Path.Combine(basePath, architecture, $"{dllFileName}.dll");
```
Comment (kasperk81) [inline src/Resolvers/Microsoft.DotNet.NativeWrapper/Interop.cs:43]:
```suggestion
            string architecture = RuntimeInformation.ProcessArchitecture.ToString().ToLowerInvariant();
            string dllPath = Path.Combine(basePath, architecture, $"{dllFileName}.dll");
```

**Comment by kasperk81** [comment]
removed `Enum.GetName`. it came up in https://github.com/dotnet/sdk/issues/15149, for each architecutre, we should use separate path. rather than hardcoding the architecture, we can use architecture provided by core library to make it future proof.

**Review by skundlony — APPROVED** [review]

**Review by marcpopMSFT — APPROVED** [review]

**Pull request merged**